Various codeAction failures in VS Code

Not sure if the root issue is with the VS Code client or with how LSP4Jakarta is handling these requests. Related LT VS Code issue: https://github.com/OpenLiberty/liberty-tools-vscode/issues/247.

First step may be to test to see if this occurs in other IDEs (Eclipse, IntelliJ)

```
May 07, 2023 5:57:19 P.M. org.eclipse.lsp4jakarta.commons.JakartaEESnippetRegistryLoader load
INFO: Loading snippets into registry...
May 07, 2023 5:57:19 P.M. org.eclipse.lsp4jakarta.JakartaLanguageServer initialize
INFO: Initializing Jakarta EE server
[Error - 6:00:11 PM] Request textDocument/codeAction failed.
  Message: The request (id: 10, method: 'textDocument/codeAction') has been cancelled
  Code: -32800 
[Error - 6:05:00 PM] Request textDocument/codeAction failed.
  Message: The request (id: 108, method: 'textDocument/codeAction') has been cancelled
  Code: -32800 
[Error - 6:05:11 PM] Request textDocument/codeAction failed.
  Message: The request (id: 130, method: 'textDocument/codeAction') has been cancelled
  Code: -32800 
[Error - 6:07:21 PM] Request textDocument/codeAction failed.
  Message: The request (id: 258, method: 'textDocument/codeAction') has been cancelled
  Code: -32800 
```

```
[Error - 6:15:15 PM] Request textDocument/codeAction failed.
  Message: Request jakarta/java/codeaction failed with message: java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
  Code: -32603
```

```
[Error - 9:53:02 AM] Request textDocument/codeAction failed.

  Message: Request jakarta/java/codeaction failed with message: java.lang.ClassCastException: class org.eclipse.jdt.core.dom.TypeDeclaration cannot be cast to class org.eclipse.jdt.core.dom.MethodDeclaration (org.eclipse.jdt.core.dom.TypeDeclaration and org.eclipse.jdt.core.dom.MethodDeclaration are in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @25107054)

  Code: -32603
```

```
May 11, 2023 9:59:00 A.M. org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation

WARNING: Unmatched cancel notification for request id 829
``` 

For this error in the "Language Support for Jakarta EE" log....
```
[Error - 6:15:15 PM] Request textDocument/codeAction failed.
  Message: Request jakarta/java/codeaction failed with message: java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
  Code: -32603
  ```
...There is a corresponding exception in the "Language Support for Java" log. On initial glance, it looks like there are multiple issues here, but definitely looks like an NPE is occurring due to the CodeActionHandler in the org.eclipse.lsp4jakarta.jdt component.
```
[Error - 6:20:09 PM] May 15, 2023, 6:20:09 p.m. BadLocationException
null
org.eclipse.jface.text.BadLocationException
	at org.eclipse.jface.text.TreeLineTracker.fail(TreeLineTracker.java:1054)
	at org.eclipse.jface.text.TreeLineTracker.offsetByLine(TreeLineTracker.java:335)
	at org.eclipse.jface.text.TreeLineTracker.getLineOffset(TreeLineTracker.java:1111)
	at org.eclipse.jface.text.AbstractLineTracker.getLineOffset(AbstractLineTracker.java:252)
	at org.eclipse.jface.text.AbstractDocument.getLineOffset(AbstractDocument.java:877)
	at org.eclipse.core.internal.filebuffers.SynchronizableDocument.getLineOffset(SynchronizableDocument.java:323)
	at org.eclipse.jdt.ls.core.internal.handlers.JsonRpcHelpers.toOffset(JsonRpcHelpers.java:85)
	at org.eclipse.jdt.ls.core.internal.handlers.JsonRpcHelpers.toOffset(JsonRpcHelpers.java:69)
	at org.eclipse.jdt.ls.core.internal.corrections.DiagnosticsHelper.getEndOffset(DiagnosticsHelper.java:38)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getProblemLocationCores(CodeActionHandler.java:304)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getCodeActionCommands(CodeActionHandler.java:138)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$13(JDTLanguageServer.java:720)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)

[Error - 10:50:10 AM] May 16, 2023, 10:50:10 a.m. Problems occurred when invoking code from plug-in: "org.eclipse.jdt.ls.core".
java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
	at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
	at org.eclipse.lsp4jakarta.jdt.internal.core.ls.JakartaDelegateCommandHandlerForJava.executeCommand(JakartaDelegateCommandHandlerForJava.java:52)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:230)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:220)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$4(JDTLanguageServer.java:585)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
	at java.base/java.util.ArrayList.addAll(Unknown Source)
	at org.eclipse.lsp4jakarta.jdt.codeAction.CodeActionHandler.codeAction(CodeActionHandler.java:143)
	at org.eclipse.lsp4jakarta.jdt.core.JDTServicesManager.getCodeAction(JDTServicesManager.java:185)
	at org.eclipse.lsp4jakarta.jdt.internal.core.ls.JakartaDelegateCommandHandlerForJava.lambda$2(JakartaDelegateCommandHandlerForJava.java:143)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.helpAsyncBlocker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.helpAsyncBlocker(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.waitingGet(Unknown Source)
	... 14 more

[Error - 10:50:10 AM] May 16, 2023, 10:50:10 a.m. Error in calling delegate command handler
java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
	at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
	at org.eclipse.lsp4jakarta.jdt.internal.core.ls.JakartaDelegateCommandHandlerForJava.executeCommand(JakartaDelegateCommandHandlerForJava.java:52)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:230)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:220)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$4(JDTLanguageServer.java:585)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "<parameter1>" is null
	at java.base/java.util.ArrayList.addAll(Unknown Source)
	at org.eclipse.lsp4jakarta.jdt.codeAction.CodeActionHandler.codeAction(CodeActionHandler.java:143)
	at org.eclipse.lsp4jakarta.jdt.core.JDTServicesManager.getCodeAction(JDTServicesManager.java:185)
	at org.eclipse.lsp4jakarta.jdt.internal.core.ls.JakartaDelegateCommandHandlerForJava.lambda$2(JakartaDelegateCommandHandlerForJava.java:143)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.helpAsyncBlocker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.helpAsyncBlocker(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.waitingGet(Unknown Source)
	... 14 more
```

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/RemoveMultipleAnnotations.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/RemoveMultipleAnnotations.java
@@ -46,6 +46,7 @@ public abstract class RemoveMultipleAnnotations extends RemoveAnnotationConflict
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         IBinding parentType = getBinding(node);
 
@@ -55,16 +56,14 @@ public abstract class RemoveMultipleAnnotations extends RemoveAnnotationConflict
                 .mapToObj(idx -> diagnosticData.get(idx).getAsString()).collect(Collectors.toList());
 
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             
             List<List<String>> annotationsListsToRemove = getMultipleRemoveAnnotations(annotations);
             for (List<String> annotationList : annotationsListsToRemove) {
                 String[] annotationsToRemove = annotationList.toArray(new String[annotationList.size()]);
                 removeAnnotation(diagnostic, context, parentType, codeActions, annotationsToRemove);
             }
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
     
     /**

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/quickfix/InsertAnnotationMissingQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/quickfix/InsertAnnotationMissingQuickFix.java
@@ -72,14 +72,13 @@ public class InsertAnnotationMissingQuickFix implements IJavaCodeActionParticipa
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         IBinding parentType = getBinding(node);
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             insertAnnotations(diagnostic, context, parentType, codeActions);
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
 
     protected static IBinding getBinding(ASTNode node) {

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/quickfix/RemoveAnnotationConflictQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/quickfix/RemoveAnnotationConflictQuickFix.java
@@ -71,15 +71,13 @@ public class RemoveAnnotationConflictQuickFix implements IJavaCodeActionParticip
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         IBinding parentType = getBinding(node);
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             removeAnnotations(diagnostic, context, parentType, codeActions);
-            return codeActions;
         }
-        return null;
-
+        return codeActions;
     }
 
     protected void removeAnnotations(Diagnostic diagnostic, JavaCodeActionContext context, IBinding parentType,

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/cdi/ManagedBeanNoArgConstructorQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/cdi/ManagedBeanNoArgConstructorQuickFix.java
@@ -44,17 +44,13 @@ public class ManagedBeanNoArgConstructorQuickFix  implements IJavaCodeActionPart
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         IBinding parentType = getBinding(node);
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
-            
             codeActions.addAll(addConstructor(diagnostic, context, parentType));
-           
-
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
     
     protected static IBinding getBinding(ASTNode node) {

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/cdi/ScopeDeclarationQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/cdi/ScopeDeclarationQuickFix.java
@@ -38,6 +38,7 @@ public class ScopeDeclarationQuickFix extends RemoveAnnotationConflictQuickFix {
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         IBinding parentType = getBinding(node);
 
@@ -49,7 +50,6 @@ public class ScopeDeclarationQuickFix extends RemoveAnnotationConflictQuickFix {
         annotations.remove(ManagedBeanConstants.PRODUCES);
 
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             /**
              * for each annotation, choose the current annotation to keep and remove the
              * rest since we can have at most one scope annotation.
@@ -62,9 +62,8 @@ public class ScopeDeclarationQuickFix extends RemoveAnnotationConflictQuickFix {
                         resultingAnnotations.toArray(new String[] {}));
             }
 
-            return codeActions;
         }
-        return null;
+        return codeActions;
 
     }
 }

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/NoResourcePublicConstructorQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/NoResourcePublicConstructorQuickFix.java
@@ -47,13 +47,13 @@ public class NoResourcePublicConstructorQuickFix implements IJavaCodeActionParti
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         MethodDeclaration parentNode = (MethodDeclaration) node.getParent();
         IMethodBinding parentMethod = parentNode.resolveBinding();
         IBinding parentType = getBinding(node);
         
         if (parentMethod != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
 
             final String TITLE_MESSAGE = "Make constructor public";
 
@@ -72,9 +72,8 @@ public class NoResourcePublicConstructorQuickFix implements IJavaCodeActionParti
             codeAction = context.convertToCodeAction(proposal, diagnostic);
             codeActions.add(codeAction);
             
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
 
     protected static IBinding getBinding(ASTNode node) {

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/NonPublicResourceMethodQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/NonPublicResourceMethodQuickFix.java
@@ -41,12 +41,12 @@ public class NonPublicResourceMethodQuickFix implements IJavaCodeActionParticipa
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         MethodDeclaration parentNode = (MethodDeclaration) node.getParent();
         IMethodBinding parentMethod = parentNode.resolveBinding();
 
         if (parentMethod != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
 
             final String TITLE_MESSAGE = "Make method public";
 
@@ -57,9 +57,8 @@ public class NonPublicResourceMethodQuickFix implements IJavaCodeActionParticipa
             CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
             codeAction.setTitle(TITLE_MESSAGE);
             codeActions.add(codeAction);
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
 
 }

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/ResourceMethodMultipleEntityParamsQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/ResourceMethodMultipleEntityParamsQuickFix.java
@@ -45,12 +45,12 @@ public class ResourceMethodMultipleEntityParamsQuickFix implements IJavaCodeActi
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         MethodDeclaration parentNode = (MethodDeclaration) node.getParent();
         IMethodBinding parentMethod = parentNode.resolveBinding();
 
         if (parentMethod != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
 
             List<SingleVariableDeclaration> params = (List<SingleVariableDeclaration>) parentNode.parameters();
 
@@ -76,9 +76,8 @@ public class ResourceMethodMultipleEntityParamsQuickFix implements IJavaCodeActi
                 codeActions.add(codeAction);
             }
 
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
 
     /**

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/persistence/PersistenceEntityQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/persistence/PersistenceEntityQuickFix.java
@@ -64,19 +64,17 @@ public class PersistenceEntityQuickFix implements IJavaCodeActionParticipant {
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         IBinding parentType = getBinding(node);
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             
             // add constructor
             if (diagnostic.getCode().getLeft().equals(PersistenceConstants.DIAGNOSTIC_CODE_MISSING_EMPTY_CONSTRUCTOR)) {
                 codeActions.addAll(addConstructor(diagnostic, context, parentType));
             }
-            
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
     
     protected static IBinding getBinding(ASTNode node) {

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/FilterImplementationQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/FilterImplementationQuickFix.java
@@ -45,10 +45,10 @@ public class FilterImplementationQuickFix implements IJavaCodeActionParticipant 
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         ITypeBinding parentType = Bindings.getBindingOfParentType(node);
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             // Create code action
             // interface
             final String TITLE_MESSAGE = "Let ''{0}'' implement ''{1}''";
@@ -60,8 +60,7 @@ public class FilterImplementationQuickFix implements IJavaCodeActionParticipant 
             CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
             codeAction.setTitle(MessageFormat.format(TITLE_MESSAGE, args));
             codeActions.add(codeAction);
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
 }

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/HttpServletQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/HttpServletQuickFix.java
@@ -45,10 +45,10 @@ public class HttpServletQuickFix implements IJavaCodeActionParticipant {
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         ITypeBinding parentType = Bindings.getBindingOfParentType(node);
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             // Create code action
             // interface
             final String TITLE_MESSAGE = "Let ''{0}'' extend ''{1}''";
@@ -60,8 +60,7 @@ public class HttpServletQuickFix implements IJavaCodeActionParticipant {
             CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
             codeAction.setTitle(MessageFormat.format(TITLE_MESSAGE, args));
             codeActions.add(codeAction);
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
 }

--- a/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/ListenerImplementationQuickFix.java
+++ b/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/ListenerImplementationQuickFix.java
@@ -45,10 +45,10 @@ public class ListenerImplementationQuickFix implements IJavaCodeActionParticipan
     @Override
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
             IProgressMonitor monitor) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
         ASTNode node = context.getCoveredNode();
         ITypeBinding parentType = Bindings.getBindingOfParentType(node);
         if (parentType != null) {
-            List<CodeAction> codeActions = new ArrayList<>();
             // Create code action
             // interface
 
@@ -77,9 +77,8 @@ public class ListenerImplementationQuickFix implements IJavaCodeActionParticipan
             codeActions.add(ca4);
             codeActions.add(ca5);
             codeActions.add(ca6);
-            return codeActions;
         }
-        return null;
+        return codeActions;
     }
 
     private CodeAction setUpCodeAction(ITypeBinding parentType, Diagnostic diagnostic, JavaCodeActionContext context,

